dev/com.ibm.ws.security/src/com/ibm/ws/security/context/SubjectRegistryManager.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    /**
+     * Set whether or not the subject is from the SAF registry.
+     *
+     * @param isSAF True if the subject is from the SAF registry.
+     */
+    public static void setSubjectIsSAF(boolean isSAF) {
+        if (!isZOS()) {
+            return;
+        }
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
+        subjectRegistryThreadContext.setIsSAF(isSAF);
+    }
+
+    /**
+     * Start subject registry detection. This is only required if a SAF
+     * registry is configured.
+     *
+     * @param isSAFRegistryConfigured
+     */
+    public static void startSubjectRegistryDetectionOnZOS() {
+        if (!isZOS()) {
+            return;
+        }
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
+        subjectRegistryThreadContext.detect();
+    }
+
+    /**
+     * Clear the subject registry detection. This is only required if a SAF
+     * registry is configured.
+     *
+     * @param isSAFRegistryConfigured
+     */
+    public static void clearSubjectRegistryDetectionOnZOS() {
+        if (!isZOS()) {
+            return;
+        }
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
+        subjectRegistryThreadContext.donotdetect();
+    }
+
+    /**
+     * Determine if a SAF credential should be created.
+     * The instances when a SAF credential should be
+     * created are:
+     * <ol>
+     * <li>SAF registry is configured and subject is from the SAF registry</li>
+     * <li>mapDistributedIdentities is true</li>
+     * <li>Subject is from the OS registry (and not logging in)</li>
+     * </ol>
+     *
+     * @return True if a SAF credential should be created and false if it should not
+     */
+    public static boolean isCreateSAFCredential() {
+        if (!isZOS()) {
+            return false;
+        }
+        SubjectRegistryThreadContext subjectRegistryThreadContext = getSubjectRegistryThreadContext();
+        return subjectRegistryThreadContext.isCreateSAFCredential();
+    }
+
+    /**
+     * Gets the thread local object.
+     * If/when a common thread storage framework is supplied, then this method
+     * implementation may need to be updated to take it into consideration.
+     *
+     * @return the thread local object.
+     */
+    @Trivial
+    private static ThreadLocal<SubjectRegistryThreadContext> getThreadLocal() {
+        return threadLocal;
+    }
+
+    /**
+     * Initialize the thread local object.
+     */
+    private static final class SecurityThreadLocal extends ThreadLocal<SubjectRegistryThreadContext> {
+        @Override
+        protected SubjectRegistryThreadContext initialValue() {
+            return new SubjectRegistryThreadContext();
+        }
+    }
+
+    /**
+     * Check if this is z/OS
+     *
+     * @return true if on z/OS
+     */
+    public static final boolean isZOS() {
```

Comment:
Why did we need to reinvent the wheel here? Can't just call the Platform.isZOS() method instead?

Reply:
(I am concerned about maintainability)